src/Phinx/Console/Command/Rollback.php

```diff
@@ -100,17 +107,66 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if (isset($envOptions['name'])) {
             $output->writeln('<info>using database</info> ' . $envOptions['name']);
         }
+        
+        $versionOrder = $this->getConfig()->getVersionOrder();
+        $output->writeln('<info>ordering by </info>' . $versionOrder);
 
         // rollback the specified environment
-        $start = microtime(true);
-        if (null !== $date) {
-            $this->getManager()->rollbackToDateTime($environment, new \DateTime($date), $force);
+        if (null === $date) {
+            $targetMustMatchVersion = true;
+            $target = $version;
         } else {
-            $this->getManager()->rollback($environment, $version, $force);
+            $targetMustMatchVersion = false;
+            $target = $this->getTargetFromDate($date);
         }
+
+        $start = microtime(true);
+        $this->getManager()->rollback($environment, $target, $force, $targetMustMatchVersion);
         $end = microtime(true);
 
         $output->writeln('');
         $output->writeln('<comment>All Done. Took ' . sprintf('%.4fs', $end - $start) . '</comment>');
     }
+
+    /**
+     * Get Target from Date
+     *
+     * @param string $date The date to convert to a target.
+     * @return The target
+     */
+    public function getTargetFromDate($date)
+    {
+        if (!preg_match('/^\d{4,14}$/', $date)) {
+            throw new \InvalidArgumentException('Invalid date. Format is YYYYmmddHHiiss (with no mandatory '.
+                'part, but the missing part(s) must all be in the end of the date string)');
```

Comment:
The format is an acceptable way of describing a parameter. It is not uncommon to see nested optional parts. Unusual to see many as this, probably, not not unusual enough to need anything more complex.

I'd recommend `Invalid value. Format is CCYY[MM[DD[HH[II[SS]]]]]`. I'd also like to have the --help page show a few examples as well as mention it is not JUST date.

-t is target (or timestamp)
-d is date (or datetime)

All a bit samey.